### PR TITLE
Adds: Skeleton classes for Category Detail Screen

### LINKED ISSUE #16058: Category Management: Add `Category Detail` screen
Parent #15549 

 - [x] #16066  
- [x] #16072 
- [x] #16087 - https://github.com/wordpress-mobile/WordPress-Android/pull/17627
- [x] #17626 - https://github.com/wordpress-mobile/WordPress-Android/pull/17627

Enhancements 
- [ ] #16114

Known issues/Bugs 
- [ ] https://github.com/wordpress-mobile/WordPress-Android/issues/17688
- [ ] While editing the category, the current parent is not shown in the parent category listing

### PR DESCRIPTION
**Pull request by AjeshRPai**
Parent #16058 

To test:
As these are only skeleton classes that are not being called in the UI these cannot be tested from the app. 

## Regression Notes
1. Potential unintended areas of impact
N/A

2. What I did to test those areas of impact (or what existing automated tests I relied on)
N/A

3. What automated tests I added (or what prevented me from doing so)
N/A

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WordPress/src/main/res/layout/category_detail_activity.xml [hunk 1]
```diff
@@ -0,0 +1,28 @@
+<?xml version="1.0" encoding="utf-8"?>
+<androidx.coordinatorlayout.widget.CoordinatorLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    android:orientation="vertical">
+
+    <fragment
```
Comment (ashiagr) [inline WordPress/src/main/res/layout/category_detail_activity.xml]: Nitpick: We might want to replace the `<fragment>` tag with `FragmentContainerView` as suggested in Android Studio.

File: WordPress/src/main/res/layout/category_detail_fragment.xml [hunk 2]
```diff
@@ -0,0 +1,11 @@
+<?xml version="1.0" encoding="utf-8"?>
+<androidx.constraintlayout.widget.ConstraintLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    xmlns:tools="http://schemas.android.com/tools"
+    android:id="@+id/category_detail"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    >
```
Comment (ashiagr) [inline WordPress/src/main/res/layout/category_detail_fragment.xml]: Nitpick: This XML could be formatted and unused namespace declarations could have been avoided.

File: WordPress/src/main/java/org/wordpress/android/ui/prefs/categories/detail/CategoryDetailFragment.kt [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+        (requireActivity().application as WordPress).component()?.inject(this)
+    }
+
+    private fun initViewModel(site: SiteModel) {
+        viewModel = ViewModelProvider(this@CategoryDetailFragment, viewModelFactory)
+                .get(CategoryDetailViewModel::class.java)
+        viewModel.start(site)
+    }
+
+    private fun getSite(savedInstanceState: Bundle?): SiteModel {
```
Comment (ashiagr) [inline WordPress/src/main/java/org/wordpress/android/ui/prefs/categories/detail/CategoryDetailFragment.kt:36]: I think we can utilize `SelectedSiteRepository` in the VM to get the currently selected site. Feel free to ignore it for now and try it in your implementation PR.

**Review by ashiagr — APPROVED** [review]
So the wait is over and you were finally able to resume this work! 😄 Can't wait to see what's coming next. 🙌 
I just left a few nitpicks, feel free to merge once you have taken a look.

File: WordPress/src/main/res/layout/category_detail_activity.xml (hunk 1 above)
Comment (AjeshRPai) [reply WordPress/src/main/res/layout/category_detail_activity.xml]: @ashiagr Yup. You are right. Didn't know about `FragmentContainerView` until now. 😅 . Done in 257b8ff

File: WordPress/src/main/res/layout/category_detail_fragment.xml (hunk 2 above)
Comment (AjeshRPai) [reply WordPress/src/main/res/layout/category_detail_fragment.xml]: @ashiagr  Yup. Have removed unused namespace and formatted the XML in 0d87b06. Nice find 👍

File: WordPress/src/main/java/org/wordpress/android/ui/prefs/categories/detail/CategoryDetailFragment.kt (hunk 3 above)
Comment (AjeshRPai) [reply WordPress/src/main/java/org/wordpress/android/ui/prefs/categories/detail/CategoryDetailFragment.kt:36]: @ashiagr Let's ignore it for now, I hadn't looked at the `SelectedSiteRepository` for getting the site model. Thanks for letting me know.  I will make that change in the implementation PR.

**Comment by AjeshRPai** [comment]
@ashiagr  Thanks for reviewing the PR and giving me suggestions. 🙌 .

**Pull request merged**